### PR TITLE
Change PVPositionerSoftDone

### LINKED ISSUE #96: How to handle done signal in PVPositioners
In the past we done something like:

```python
def move(self, *args, **kwargs):
    self.done.put(0)
    return super().move(*args, **kwargs)
```

The above doesn't work well for some cases, like the undulatory. So it looks like the best would be:
```python
def move(self, *args, **kwargs):
    status = super().move(*args, **kwargs)
    self.done.get()
    return status
```

but this needs to be tested.

### LINKED ISSUE #101: Test Pete's version of a PVPositioner with soft done
https://github.com/BCDA-APS/bluesky_instrument_training/blob/main/instrument/devices/temperature_signal.py

It does the same as ours, but keeps all parts in the `PVPositioner` class.

### PR DESCRIPTION
**Pull request by gfabbris**
Changes all `PVPositionerSoftDone` to a modified version from @prjemian (https://github.com/BCDA-APS/bluesky_instrument_training/blob/main/instrument/devices/temperature_signal.py), which makes the code much simpler.

There are two main differences from Pete's version:
- Option to pass the `setpoint` and `readback` PVs in the `__init__`.
- Option to pass a `target_attr`. This is important for positioners that has ramping because if the `setpoint` is slowly changed by EPICS, then `readback.get() - setpoint.get()` might go to zero, and Bluesky things that the motion is done. The solution here is to create a `target` signal that will just hold to intended final position. Our lakeshore is a good example of this: 
https://github.com/APS-4ID-POLAR/ipython-polar/blob/6832cc2184dc3f03db01cbc0a4153aef90bf4ede/profile_bluesky/startup/instrument/devices/lakeshore340.py#L60

Close #96, closes #101.

### CODE REVIEW

File: profile_bluesky/startup/instrument/devices/aps_source.py [hunk 1]
```diff
@@ -22,7 +22,6 @@
 class UndulatorEnergy(PVPositioner):
     """
     Undulator energy positioner.
-
```
Comment (prjemian) [inline profile_bluesky/startup/instrument/devices/aps_source.py:25]: This vertical whitespace is a good practice.  When using Sphinx, it separates the summary line (used by [autosummary](https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html?highlight=autosummary)) from the remainder of the documentation content.  Is there a need to remove it from here (and other places in thie PR)?

File: profile_bluesky/startup/instrument/devices/aps_source.py [hunk 2]
```diff
@@ -22,7 +22,6 @@
...
     Main purpose this being a PVPositioner is to handle the undulator backlash.
```
Comment (prjemian) [inline profile_bluesky/startup/instrument/devices/aps_source.py:25]: replace with this text (and resolve the WHICH_WAY):
`Always move the undulator to final position from the WHICH_WAY direction, by applying a backlash (hysteresis) correction as needed.`

File: profile_bluesky/startup/instrument/devices/ge_controller.py [hunk 3]
```diff
@@ -4,53 +4,29 @@
...
-
-    slew = FormattedComponent(EpicsSignalWithRBV, "{self.prefix}Slew",
-                              kind=Kind.config)
-
-    effort = FormattedComponent(EpicsSignalRO, "{self.prefix}Effort_RBV",
-                                auto_monitor=True, kind=Kind.config)
-
-    def __init__(self, *args, loop_number=None, timeout=60*60*10, **kwargs):
-        self.loop_number = loop_number
+    units = Component(EpicsSignalWithRBV, "{self.prefix}Units", kind="config")
```
Comment (prjemian) [inline profile_bluesky/startup/instrument/devices/ge_controller.py:18]: Isn't `{self.prefix}` applied by `Component.__init__()` as a default?

File: profile_bluesky/startup/instrument/devices/ge_controller.py [hunk 4]
```diff
@@ -84,25 +48,15 @@ def stop(self, *, success=False):
             self.setpoint.put(self._position)
         super().stop(success=success)
```
Comment (prjemian) [inline profile_bluesky/startup/instrument/devices/ge_controller.py:49]: Why stop here?

Comment (prjemian) [reply profile_bluesky/startup/instrument/devices/ge_controller.py:49]: The GitHub diffs make it look as if the stop here is in the egu property above.

File: profile_bluesky/startup/instrument/devices/high_field_magnet.py [hunk 5]
```diff
@@ -4,137 +4,93 @@
...
     ramp_rate = Component(EpicsSignal, "Rate", write_pv="RateSet",
-                          kind=Kind.config)
+                          kind="config")
 
 
 class AMIController(PVPositioner):
     """ Magnetic field controler """
 
     # position
     readback = Component(EpicsSignalRO, "Field", auto_monitor=True,
```
Comment (prjemian) [inline profile_bluesky/startup/instrument/devices/high_field_magnet.py:25]: Don't know if you saw it with the EpicsMotor but I like the way that is configured.  In the `__init__()` method, after it calls `super().__init__(...)`,

```python
self.readback.name = self.name
```

so this makes the `.read()` method report `motor=..., motor.setpoint=...` where the first value is actually `motor.readback`.  The column labels are simpler then.  You might consider that here.

**Review by prjemian — COMMENTED** [review]
Good job, some comments with changes.  I suggest you consider applying [`black`](https://black.readthedocs.io/en/stable/index.html) to these files to apply a consistent code appearance (indentation, splitting of long argument lists, ...).  Reviews become easier when the code style is not a distraction.  Indeed, black is too opinionated but it is *very* consistent.

File: profile_bluesky/startup/instrument/devices/aps_source.py (hunk 1 above)
Comment (gfabbris) [reply profile_bluesky/startup/instrument/devices/aps_source.py:25]: I'm not sure I understand this comment. Shouldn't I add a blank line to make it better for Sphinx? I think it doesn't like if the first like is too long.

File: profile_bluesky/startup/instrument/devices/ge_controller.py (hunk 3 above)
Comment (gfabbris) [reply profile_bluesky/startup/instrument/devices/ge_controller.py:18]: Yeah! I had them all as `FormattedComponent` before...

File: profile_bluesky/startup/instrument/devices/aps_source.py (hunk 1 above)
Comment (prjemian) [reply profile_bluesky/startup/instrument/devices/aps_source.py:25]: Right.  You *want* the blank line.  Github is showing that the blank line has been removed.

File: profile_bluesky/startup/instrument/devices/aps_source.py [hunk 6]
```diff
@@ -108,7 +114,7 @@ def _move(self, position, **kwargs):
         """
         Moves undulator.
 
-        This is meant to the ran using threading, so the move will block by
+        This is meant to the be ran using threading, so the move will block by
```
Comment (prjemian) [inline profile_bluesky/startup/instrument/devices/aps_source.py:117]: change to: `This is meant to run using threading, so the move will block by`

**Review by prjemian — COMMENTED** [review]
Nice!  One change of wording and we're good to go.

File: profile_bluesky/startup/instrument/devices/high_field_magnet.py (hunk 5 above)
Comment (gfabbris) [reply profile_bluesky/startup/instrument/devices/high_field_magnet.py:25]: This is a good one. Should this be also the standard for `ophyd.PVPositioner`?

Comment (prjemian) [reply profile_bluesky/startup/instrument/devices/high_field_magnet.py:25]: My opinion is yes.  But it is a preference.

**Review by prjemian — APPROVED** [review]

**Pull request merged**